pkg/planner/core/integration_test.go

```diff
@@ -2408,3 +2408,120 @@ func TestAggregationInWindowFunctionPushDownToTiFlash(t *testing.T) {
 		tk.MustQuery("explain select sum(v) over w as res1, count(v) over w as res2, avg(v) over w as res3, min(v) over w as res4, max(v) over w as res5 from t window w as (partition by p order by o);").CheckAt([]int{0, 2, 4}, rows)
 	})
 }
+func TestIssuenullreject(t *testing.T) {
+	testkit.RunTestUnderCascades(t, func(t *testing.T, tk *testkit.TestKit, cascades, caller string) {
```

Comment:
We have many issues which is related with `isNullRejected`. Can you check their PR and then to close or ref them?

Reply:
For NULL AND (1+2)=3:

- (1+2)=3 is TRUE.

- NULL AND TRUE is NULL under three-valued logic (3VL).

In a WHERE clause, NULL is treated as FALSE, so it rejects all rows.